src/api.ts

```diff
@@ -0,0 +1,187 @@
+    }
+  });
+
+  // Get bundles from file tokens
+  bundles.push(...getBundles(fileTokens));
+
+  return Promise.all(
+    bundles.map(bundle =>
+      exploreBundle(bundle, options).catch<ExploreErrorResult>(error =>
+        onExploreError(bundle, error)
+      )
+    )
+  )
+    .then(results => getExploreResult(results, options))
+    .then(result => {
+      // Reject if none of results is successful
+      if (result.bundles.length === 0) {
+        return Promise.reject(result);
+      }
+
+      saveHtmlToFile(result, options);
+
+      return result;
+    });
+}
+
+/**
+ * Expand list of file tokens into a list of bundles
+ */
+export function getBundles(fileTokens: string[]): Bundle[] {
+  const filenames = fileTokens.reduce<string[]>((result, filePath) => {
+    if (glob.hasMagic(filePath)) {
+      result.push(...expandGlob(filePath));
+    } else {
+      result.push(filePath);
+    }
+
+    return result;
+  }, []);
+
+  const codeFilenames: string[] = [];
+  const mapFilenames: string[] = [];
+
+  filenames.forEach(filename => {
+    if (filename.endsWith('.map')) {
+      mapFilenames.push(filename);
+    } else {
+      codeFilenames.push(filename);
+    }
+  });
+
+  return codeFilenames.map<Bundle>(code => ({
+    code,
+    map: mapFilenames.find(filename => filename === `${code}.map`),
+  }));
+}
+
+function expandGlob(pattern: string): string[] {
+  // Make sure pattern match `.map` files as well
+  if (pattern.endsWith('.js')) {
+    pattern = `${pattern}?(.map)`;
+  }
+
+  return glob.sync(pattern);
+}
+
+export function getBundleName(bundle: Bundle): string {
+  return Buffer.isBuffer(bundle.code) ? 'Buffer' : bundle.code;
+}
+
+/**
+ * Handle error during bundle processing
+ */
+function onExploreError(bundle: Bundle, error: NodeJS.ErrnoException): ExploreErrorResult {
+  return {
+    bundleName: getBundleName(bundle),
+    code: error.code || ErrorCode.Unknown,
+    message: error.message,
+    error,
+  };
+}
+
+function getExploreResult(
+  results: (ExploreBundleResult | ExploreErrorResult)[],
+  options: ExploreOptions
+): ExploreResult {
+  const bundles: ExploreBundleResult[] = [];
+  const errors: ExploreErrorResult[] = [];
+
+  results.forEach(result => {
+    if ((result as ExploreBundleResult).files) {
+      bundles.push(result as ExploreBundleResult);
+    } else {
+      errors.push(result as ExploreErrorResult);
+    }
+  });
+
+  errors.push(...getPostExploreErrors(bundles));
+
+  return {
+    bundles,
+    errors,
+    ...((options.html || options.file) && { html: generateHtml(bundles) }),
+  };
+}
+
+function getPostExploreErrors(exploreBundleResult: ExploreBundleResult[]): ExploreErrorResult[] {
+  return exploreBundleResult.reduce<ExploreErrorResult[]>((result, exploreBundleResult) => {
+    const { bundleName, files, totalBytes } = exploreBundleResult;
+
+    const filenames = Object.keys(files).filter(filename => filename !== UNMAPPED_KEY);
+    if (filenames.length === 1) {
```

Comment:
What if there are multiple filenames? Worth a comment at least?

Reply:
There was an error message explaining why only one file is an error. It's now hidden inside `getErrorMessage(ErrorCode.OneSourceSourceMap`. But I'll add a comment

Reply:
Comment added